circuit_knitting/cutting/cutting_decomposition.py

```diff
@@ -28,17 +28,24 @@
 from ..utils.observable_grouping import observables_restricted_to_subsystem
 from ..utils.transforms import separate_circuit
 from .qpd.qpd_basis import QPDBasis
-from .qpd.instructions import TwoQubitQPDGate
+from .qpd.instructions import BaseQPDGate, SingleQubitQPDGate, TwoQubitQPDGate
 
 
 class PartitionedCuttingProblem(NamedTuple):
     """The result of decomposing and separating a circuit and observable(s)."""
 
     subcircuits: dict[str | int, QuantumCircuit]
-    bases: list[QPDBasis]
+    cuts: list[CutInfo]
     subobservables: dict[str | int, QuantumCircuit] | None = None
 
 
+class CutInfo(NamedTuple):
+    """The decomposition and location information associated with one cut."""
+
+    basis: QPDBasis
+    gates: list[tuple[Hashable, int]] | list[int] | int
```

Comment:
This data tuple can be used to describe multiple gates spread across several subcircuits, multiple gates within a single circuit, and one gate within a single circuit